Signal/src/call/CallService.swift

```diff
@@ -1152,6 +1161,33 @@ protocol CallServiceObserver: class {
         // Do nothing
     }
 
+    internal func callRecordDidChange(call: SignalCall, callRecord: TSCall?) {
+        AssertIsOnMainThread()
+
+        updateCallRecordType(call: call)
+    }
+
+    internal func updateCallRecordType(call: SignalCall?) {
```

Comment:
nit: do you dare move this to something like `call.completeAnyCallRecord()` method. This class is so effing huge. 

Also I think that would let us get rid of the `callRecordDidChange` machinery which is adding a few lines of code for how little it's doing.